baseline_module/config_backup.txt
# configuration of baseline models (classification)
import random
import numpy as np
import torch

absolute_path_prefix = r'/home/jsjlab/projects/AttackViaGan'

baseline_debug_mode = False


def setup_seed(seed):
    print(f'setup seed {seed}')
    torch.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)
    np.random.seed(seed)
    random.seed(seed)
    torch.backends.cudnn.deterministic = True

setup_seed(667)

baseline_config_model_save_path_format = absolute_path_prefix+r'/baseline_module/baseline_models/{}/{}_{:.5f}_{}_{}.pt'

baseline_config_model_load_path = {
    'IMDB': {
        'LSTM': absolute_path_prefix+r'/baseline_module/baseline_models/IMDB/LSTM_0.86532_01_24-13-05_.pt',
        'BidLSTM': absolute_path_prefix+r'/baseline_module/baseline_models/IMDB/BidLSTM_0.86444_01_24-14-12_.pt',
        'TextCNN': absolute_path_prefix+r'/baseline_module/baseline_models/IMDB/TextCNN_0.86736_01_24-15-01_.pt',
        'Bert': absolute_path_prefix+r'/baseline_module/baseline_models/IMDB/Bert_0.91652_01_23-23-59_.pt',
    },
    'AGNEWS': {
        'LSTM': absolute_path_prefix+r'/baseline_module/baseline_models/AGNEWS/LSTM_0.90118_.pt',
        'BidLSTM': absolute_path_prefix+r'/baseline_module/baseline_models/AGNEWS/BidLSTM_0.91855_.pt',
        'TextCNN': absolute_path_prefix+r'/baseline_module/baseline_models/AGNEWS/TextCNN_0.91803_.pt',
        'Bert': absolute_path_prefix+r'/baseline_module/baseline_models/AGNEWS/Bert_0.94132_01_24-01-24_.pt',
    },
    'SNLI': {
        'LSTM_E': absolute_path_prefix+r'/baseline_module/baseline_models/SNLI/LSTM_E_0.74338_01_29-23-40_.pt',
        'BidLSTM_E': absolute_path_prefix+r'/baseline_module/baseline_models/SNLI/BidLSTM_E_0.77769_03_05-21-53_.pt',
        'TextCNN_E': absolute_path_prefix+r'/baseline_module/baseline_models/SNLI/TextCNN_E_0.74287_01_29-14-46_.pt',
        'Bert_E': absolute_path_prefix+r'/baseline_module/baseline_models/SNLI/Bert.pt',
    }
}

class baseline_IMDBConfig():
    train_data_path = absolute_path_prefix+r'/dataset/IMDB/aclImdb/train.std'
    test_data_path = absolute_path_prefix+'/dataset/IMDB/aclImdb/test.std'
    pretrained_word_vectors_path = absolute_path_prefix+r'/static/glove.6B.100d.txt'
    labels_num = 2
    vocab_limit_size = 40000
    tokenizer_type = 'normal'
    padding_maxlen = 230

class baseline_AGNEWSConfig():
    train_data_path = absolute_path_prefix+r'/dataset/AGNEWS/train.std'
    test_data_path = absolute_path_prefix+r'/dataset/AGNEWS/test.std'
    pretrained_word_vectors_path = absolute_path_prefix+r'/static/glove.6B.100d.txt'
    labels_num = 4
    vocab_limit_size = 80000
    tokenizer_type = 'normal'
    padding_maxlen = 50

class baseline_SNLIConfig():
    train_data_path = absolute_path_prefix+r'/dataset/SNLI/train.txt'
    test_data_path = absolute_path_prefix+r'/dataset/SNLI/test.txt'
    sentences_path = absolute_path_prefix+r'/dataset/SNLI/sentences.txt'
    pretrained_word_vectors_path = absolute_path_prefix+r'/static/glove.840B.300d.txt'
    labels_num = 3
    vocab_limit_size = 80000
    tokenizer_type = 'spacy'
    padding_maxlen = 15

class baseline_TextCNNConfig():
    channel_kernel_size = {
        'IMDB': ([50, 50, 50], [3, 4, 5]),
        'AGNEWS': ([50, 50, 50], [3, 4, 5]),
        'SNLI': ([30, 40, 50, 60], [2, 3, 4, 5])
    }
    is_static = {
        'IMDB': True,
        'AGNEWS': True,
        'SNLI': True,
    }
    using_pretrained = {
        'IMDB': True,
        'AGNEWS': True,
        'SNLI': True
    }

    train_embedding_dim = {
        'IMDB': 50,
        'AGNEWS': 50,
        'SNLI': 100
    }

    is_batch_normal = {
        'IMDB': True,
        'AGNEWS': False,
        'SNLI': True
    }

class baseline_LSTMConfig():
    num_hiddens = {
        'IMDB': 128,
        'AGNEWS': 100,
        'SNLI': 2048,
    }

    num_layers = {
        'IMDB': 2,
        'AGNEWS': 2,
        'SNLI': 1,
    }

    is_using_pretrained = {
        'IMDB': False,
        'AGNEWS': True,
        'SNLI': True,
    }

    word_dim = {
        'IMDB': 50,
        'AGNEWS': 100,
        'SNLI': 300
    }

    is_head_and_tail = {
        'IMDB': False,
        'AGNEWS': True,
        'SNLI': True
    }

    linear_size = {
        'SNLI': 500,
    }

    dropout_rate = {
        'SNLI': 0.0,
    }

    pool_type = {
        'SNLI': 'no',
    }


class baseline_BertConfig():
    model_name = 'bert-base-uncased'
    is_fine_tuning = {
        'IMDB': True,
        'AGNEWS': True,
        'SNLI': True
    }

baseline_config_dataset = {
    'IMDB': baseline_IMDBConfig,
    'AGNEWS': baseline_AGNEWSConfig,
    'SNLI': baseline_SNLIConfig,
}

baseline_config_models_list = [
    'Bert', 'LSTM', 'BidLSTM', 'TextCNN',
    'Bert_E', 'LSTM_E', 'BidLSTM_E', 'TextCNN_E'
]


config.py
import torch

bert_vocab_size = 30522
glove_embedding_size = 100
config_path = './config.py'


class AttackConfig():
    output_dir = r'./output'
    train_multi_cuda = False
    cuda_idx = 1
    if train_multi_cuda:
        multi_cuda_idx = [0, 1, 2, 3]
        cuda_idx = multi_cuda_idx[0]
    train_device = torch.device('cuda:' + str(cuda_idx))
    dataset = 'IMDB'  # choices = 'IMDB', 'AGNEWS', 'SNLI'
    baseline_model = 'LSTM'  # choices = 'LSTM', 'TextCNN', 'BidLSTM', 'Bert'
    debug_mode = False
    epochs = 120
    batch_size = 128

    load_pretrained_Seq2Seq = True
    Seq2Seq_BidLSTM = False
    head_tail = False
    fine_tuning = False

    if load_pretrained_Seq2Seq:
        if dataset == 'AGNEWS':
            pretrained_Seq2Seq_path = r'./output/seq2seq_model/AGNEWS/1611484301/Seq2Seq_model.pt'
        if dataset == 'IMDB':
            pretrained_Seq2Seq_path = r'./output/seq2seq_model/IMDB/1614672817/Seq2Seq_model.pt'
        if dataset == 'SNLI':
            if not Seq2Seq_BidLSTM:
                pretrained_Seq2Seq_path = r'./output/seq2seq_model/SNLI/1615217103/Seq2Seq_model.pt'
            else:
                pretrained_Seq2Seq_path = r'./output/seq2seq_model/SNLI/1615217989/Seq2Seq_model.pt'
    if fine_tuning:
        Seq2Seq_learning_rate_BERT = 5e-6
        Seq2Seq_learning_rate_LSTM = 1e-4
    else:
        Seq2Seq_learning_rate = 1e-3
    gan_gen_learning_rate = 1e-4
    gan_adv_learning_rate = 1e-4

    hidden_size = 768
    num_layers = 3
    dropout = 0.3
    super_hidden_size = 500
    vocab_size = bert_vocab_size

    gan_schedule = [1, 3, 5]
    seq2seq_train_times = 1
    gan_gen_train_times = 1
    gan_adv_train_times = 5
    gan_gen_train_limit = epochs

    perturb_sample_num = 10
    perturb_search_times = 10
    perturb_search_bound = 0.16
    perturb_mul = 1.16


class BaselineConfig():
    output_dir = r'./output'
    cuda_idx = 1
    train_device = torch.device('cuda:' + str(cuda_idx))
    dataset = 'AGNEWS'  # choices = 'IMDB', 'AGNEWS', 'SNLI'
    baseline_model = 'LSTM'  # choices = 'LSTM', 'TextCNN', 'BidLSTM', 'Bert'
    debug_mode = False
    epochs = 20
    batch_size = 64


class IMDBConfig():
    train_data_path = r'./dataset/IMDB/aclImdb/train'
    test_data_path = r'./dataset/IMDB/aclImdb/test'
    labels_num = 2
    tokenizer_type = 'bert'
    remove_stop_words = False
    sen_len = 230
    vocab_size = bert_vocab_size


class AGNEWSConfig():
    train_data_path = r'./dataset/AGNEWS/train.std'
    test_data_path = r'./dataset/AGNEWS/test.std'
    labels_num = 4
    tokenizer_type = 'bert'
    remove_stop_words = False
    sen_len = 50
    vocab_size = bert_vocab_size


class SNLIConfig():
    train_data_path = r'./dataset/SNLI/train.txt'
    test_data_path = r'./dataset/SNLI/test.txt'
    sentences_data_path = r'./dataset/SNLI/sentences.txt'
    labels_num = 3
    label_classes = {'neutral': 0, 'contradiction': 1, 'entailment': 2}
    tokenizer_type = 'bert'
    remove_stop_words = False
    sen_len = 15
    vocab_size = bert_vocab_size


class Baseline_LSTMConfig():
    vocab_size = bert_vocab_size
    embedding_size = glove_embedding_size
    vocab_path = r'./static/vocab.txt'
    embedding_path = r'./static/glove.6B.100d.txt'
    hidden_size = 100
    num_layers = 2
    using_pretrained = True
    head_tail = True
    dropout = 0.3
    learning_rate = {'IMDB': 1e-3, 'AGNEWS': 1e-3, 'SNLI': 1e-3}


class Baseline_CNNConfig():
    vocab_size = bert_vocab_size
    embedding_size = glove_embedding_size
    vocab_path = r'./static/vocab.txt'
    embedding_path = r'./static/glove.6B.100d.txt'
    channel_size = [200, 200, 200]
    kernel_size = [3, 4, 5]
    using_pretrained = True
    dropout = 0.3
    learning_rate = {'IMDB': 1e-3, 'AGNEWS': 1e-3, 'SNLI': 1e-3}


class Baseline_BertConfig():
    vocab_size = bert_vocab_size
    hidden_size = 768
    fine_tuning = False
    learning_rate = {'IMDB': 1e-3, 'AGNEWS': 1e-3, 'SNLI': 1e-3}


dataset_config_data = {
    'IMDB': IMDBConfig,
    'AGNEWS': AGNEWSConfig,
    'SNLI': SNLIConfig,
}

baseline_model_config_data = {
    'LSTM': Baseline_LSTMConfig,
    'BidLSTM': Baseline_LSTMConfig,
    'TextCNN': Baseline_CNNConfig,
    'Bert': Baseline_BertConfig
}

baseline_model_lists = ['LSTM', 'TextCNN', 'BidLSTM', 'Bert']

dataset_list = [
    'IMDB',
    'AGNEWS',
    'SNLI',
]

baseline_model_load_path = {
    'IMDB': {
        'LSTM':
        r'./output/baseline_model/IMDB/LSTM/1611247993/baseline_model.pt',
        'TextCNN':
        r'./output/baseline_model/IMDB/TextCNN/1611248030/baseline_model.pt',
        'BidLSTM':
        r'./output/baseline_model/IMDB/BidLSTM/1611248060/baseline_model.pt',
        'Bert':
        r'./output/baseline_model/IMDB/Bert/1611248987/baseline_model.pt',
    },
    'AGNEWS': {
        'LSTM':
        r'./output/baseline_model/AGNEWS/LSTM/1611246841/baseline_model.pt',
        'BidLSTM':
        r'./output/baseline_model/AGNEWS/BidLSTM/1611246875/baseline_model.pt',
        'TextCNN':
        r'./output/baseline_model/AGNEWS/TextCNN/1611246902/baseline_model.pt',
        'Bert':
        r'./output/baseline_model/AGNEWS/Bert/1611246760/baseline_model.pt',
    },
    'SNLI': {}
}
